app/models/preupgrade_report.rb

```diff
@@ -7,12 +7,17 @@ class PreupgradeReport < ::Report
   scoped_search on: :job_invocation_id, only_explicit: true
 
   def self.create_report(host, data, job_invocation_id)
-    report = PreupgradeReport.create(host: host, status: 0,
-                                     job_invocation_id: job_invocation_id,
-                                     reported_at: DateTime.now.utc)
+    # We don't have specific permissions for the Preupgrade leapp reports,
+    # so we need to skip the permission check for non-admin users.
+    # The user is still required to have permission to run the job and view the hosts.
+    skip_permission_check do
```

Comment:
I tried other methods from `authorizable.rb`, like `authorized_as`, `joins_authorized_as`, but nothing worked, only `skip_permission_check`.

I think it's safe in this case because the user's permissions are checked in other parts of the application (view hosts, run jobs etc) prior this